core-common/src/main/java/org/zalando/nakadi/service/subscription/model/Partition.java

```diff
@@ -52,8 +56,47 @@ public Partition(
         this.state = state;
     }
 
+    public Partition(
+            final String eventType,
+            final String partition,
+            @Nullable final String session,
+            @Nullable final String nextSession,
+            final State state,
+            final int failedCommitsCount,
+            final String lastDeadLetterOffset) {
+        this.eventType = eventType;
+        this.partition = partition;
+        this.session = session;
+        this.nextSession = nextSession;
+        this.state = state;
+        this.failedCommitsCount = failedCommitsCount;
+        this.lastDeadLetterOffset = lastDeadLetterOffset;
+    }
+
     public Partition toState(final State state, @Nullable final String session, @Nullable final String nextSession) {
-        return new Partition(eventType, partition, session, nextSession, state);
+        return new Partition(eventType, partition, session, nextSession,
+                state, failedCommitsCount, lastDeadLetterOffset);
+    }
+
+    public Partition toPartitionWithIncFailedCommits() {
+        return new Partition(eventType, partition, session, nextSession,
+                state, failedCommitsCount + 1, lastDeadLetterOffset);
+    }
+
+    public Partition toZeroFailedCommits() {
+        return new Partition(eventType, partition, session, nextSession, state, 0, lastDeadLetterOffset);
+    }
+
+    public Partition toLookingDeadLetter(final String lastDeadLetterOffset) {
+
+        if (lastDeadLetterOffset != null && this.lastDeadLetterOffset == null) {
+            // failed commits reset to count for failures of the specific event
+            return new Partition(eventType, partition, session, nextSession, state, 0, lastDeadLetterOffset);
+        } else if (lastDeadLetterOffset == null && this.lastDeadLetterOffset != null) {
+            return new Partition(eventType, partition, session, nextSession, state, 0, null);
+        }
```

Comment:
```suggestion
        if ((lastDeadLetterOffset != null && this.lastDeadLetterOffset == null) || (lastDeadLetterOffset == null && this.lastDeadLetterOffset != null)) {
            return new Partition(eventType, partition, session, nextSession, state, 0, lastDeadLetterOffset);
        }
```
In both cases we reset failed commit count to 0, and use the given value of `lastDeadLetterOffset` (either it is `null` or not).